dart-templates/serialization/native/class.mustache

```diff
@@ -281,7 +288,13 @@ class {{{classname}}} {{#interfaces}}{{#-first}}implements {{/-first}}{{.}}{{^-l
             {{name}},
           {{/isContainer}}
           {{^isContainer}}
-            {{name}}{{^required}}?{{/required}}.toJson(),
+            {{name}}{{^required}}?{{/required}}.toJson(keys?.fold<List<String>>(<String>[], (List<String> previousValue, String element) {
+              if (element.startsWith(r'{{{name}}}.')) {
+                previousValue.add(element.split(r'{{{name}}}.').last);
```

Comment:
Just wondering if we could have edge cases where a key is redondant (eg 'links.entity.links...') but it might be nothing to worry about